Dockerfile

```diff
@@ -85,14 +85,134 @@ RUN adduser airflow \
     && echo "airflow ALL=(ALL) NOPASSWD: ALL" > /etc/sudoers.d/airflow \
     && chmod 0440 /etc/sudoers.d/airflow
 
+############################################################################################################
+# This is an image with all APT dependencies needed by CI. It is built on top of the airlfow APT image
+# Parameters:
+#     airflow-apt-deps - this is the base image for CI deps image.
+############################################################################################################
+FROM airflow-apt-deps as airflow-ci-apt-deps
+
+SHELL ["/bin/bash", "-o", "pipefail", "-e", "-u", "-x", "-c"]
+
+ENV JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/
+
+ARG APT_DEPS_IMAGE
+ENV APT_DEPS_IMAGE=${APT_DEPS_IMAGE}
+
+RUN echo "${APT_DEPS_IMAGE}"
+
+# Note the ifs below might be removed if Buildkit will become usable. It should skip building this
+# image automatically if it is not used. For now we still go through all layers below but they are empty
+# Note missing directories on debian-stretch https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=863199
+RUN if [[ "${APT_DEPS_IMAGE}" == "airflow-ci-apt-deps" ]]; then \
+        mkdir -pv /usr/share/man/man1 \
+        && mkdir -pv /usr/share/man/man7 \
+        && apt-get update \
+        && apt-get install --no-install-recommends -y \
+          lsb-release gnupg dirmngr openjdk-8-jdk \
+          vim tmux less unzip net-tools netcat \
+          ldap-utils postgresql-client sqlite3 \
+          krb5-user openssh-client openssh-server \
+          python-selinux \
```

Comment:
No mysql-client?

Reply:
Well.. It's installed few steps below. That was one of the differences I had to battle - in debian-stretch installing mysql from apt-get results in installing mariadb. Which is fine being open-source and if you do not like Oracle ;). But I guess the goal is to test it with real Oracle^H^H^H^H^H^HMysql client (unless we are fine with mariadb :) )

Actually that installation method might still change as I still did not have mysql running on Travis (it works locally) so I presume something with the installation method messes it up - likely location of some configuration directories.

Reply:
For client libs I don't see a problem using the debian provided libs, even if those are maraidb: That is what a number of people would get if they aren't running in docker.

Reply:
OK. That's good then. It will be easier a bit then :). I will test it as i am close to get simpler mysql install following the other problems.

Reply:
I looked into it and I think we should stick with MySQL. It's only pure Debian Stretch that released MariaDB packages as default (and only) choice: https://mariadb.org/debian-9-released-mariadb-mysql-variant/ . On Ubuntu 18.04, MySQL is still the default choice and you have to add it manually (https://tecadmin.net/install-mariadb-on-ubuntu-18-04-bionic/). I am not 100% sure, but I think Ubuntu user base will be far bigger than Debian's so having a "true" MySQL libs makes perfect sense. MySQL is still FOSS so no problems with licensing etc - and I am using official Oracle installation instructions for Debian, so I guess this should be fine @ashb?

Reply:
I think we're having this conversation again in another PR, right? We agreed (already changed?) the base image to use os provided libmysqlclient packages right?

Reply:
Right. Changing it now back to use mariadb libraries. Will make it a bit smaller and faster :)